api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -59,46 +61,30 @@ public AbstractHandler() {
 	 * @return File that the complex data should be written to
 	 */
 	public File getOutputFileToWrite(Obs obs) throws IOException {
-		// Get the title and remove the extension.
+		// Get the title and the extension
 		String title = obs.getComplexData().getTitle();
-		String extension = "." + getExtension(title);
-		
-		// If getExtension returns the title, there was no extension
-		if (getExtension(title).equals(title)) {
-			extension = "";
-		}
+		String extension = FilenameUtils.getExtension(title);
 		
 		File dir = OpenmrsUtil.getDirectoryInApplicationDataDirectory(Context.getAdministrationService().getGlobalProperty(
 		    OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR));
-		File outputfile;
+		File outputfile = null;
 		
-		// Get the output stream
+		// Generate the filename with the Obs' UUID
+		String uuid = obs.getUuid();
 		if (null == title) {
 			String now = longfmt.format(new Date());
-			outputfile = new File(dir, now);
+			outputfile = new File(dir, now + '_' + uuid);
 		} else {
-			title = title.replace(extension, "");
+			// If the extension is no empty string, it exists and we can add a dot in front for later use
+			if (!(extension.equals(""))) {
```

Comment:
I think it is always safer to do the reverse (just an hedge against NPE):
```java
if (!("".equals(extension)))
```
Or simply
```java
if (!StringUtils.isEmpty(extension))
```

Reply:
the latter one is safer when you want to exclude inputs like `"    "`